Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +351,299 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+        average time, measured over ``nrepeats`` invocations, spent in
+        the python extension. ``sigma_time`` is the dispersion of the
+        run times across those ``nrepeats`` invocations.
+
+    Example
+    -------
+    >>> from __future__ import print_function
+    >>> import numpy as np
+    >>> import time
+    >>> from math import pi
+    >>> from os.path import dirname, abspath, join as pjoin
+    >>> import Corrfunc
+    >>> from Corrfunc.mocks.DDtheta_mocks \
+        import find_fastest_DDtheta_mocks_bin_refs
+    >>> binfile = pjoin(dirname(abspath(Corrfunc.__file__)),
+    ...                 "../mocks/tests/", "angular_bins")
+    >>> N = 100000
+    >>> nthreads = 4
+    >>> seed = 42
+    >>> np.random.seed(seed)
+    >>> RA1 = np.random.uniform(0.0, 2.0*pi, N)*180.0/pi
+    >>> cos_theta = np.random.uniform(-1.0, 1.0, N)
+    >>> DEC1 = 90.0 - np.arccos(cos_theta)*180.0/pi
+    >>> autocorr = 1
+    >>> best, _ = find_fastest_DDtheta_mocks_bin_refs(autocorr, nthreads,\
+    ...                                               binfile, RA1, DEC1,\
+    ...                                               return_runtimes=True)
+    >>> print(best) # doctest:+SKIP
+    (2, 1)
+
+    .. note:: Since the result might change depending on the computer,
+    doctest is skipped for this function.
+    """
+
+    weights1 = None
+    weights2 = None
+    weight_type = None
+
+    try:
+        from Corrfunc._countpairs_mocks import countpairs_theta_mocks as\
+            DDtheta_mocks_extn
+    except ImportError:
+        msg = "Could not import the C extension for the angular "\
+              "correlation function for mocks."
+        raise ImportError(msg)
+
+    import numpy as np
+    from Corrfunc.utils import translate_isa_string_to_enum, fix_ra_dec,\
+        return_file_with_rbins, convert_to_native_endian, process_weights
+    from future.utils import bytes_to_native_str
+    import itertools
+    import time
+
+    weights1, weights2 = process_weights(weights1, weights2, RA1, RA2,
+                                         weight_type, autocorr)
+
+    # Ensure all input arrays are native endian
+    RA1, DEC1, weights1, RA2, DEC2, weights2 = [
+        convert_to_native_endian(arr, warn=True) for arr in
+        [RA1, DEC1, weights1, RA2, DEC2, weights2]]
+
+    fix_ra_dec(RA1, DEC1)
+    if autocorr == 0:
+        fix_ra_dec(RA2, DEC2)
+
+    # Passing None parameters breaks the parsing code, so avoid this
+    kwargs = {}
+    for k in ['weights1', 'weights2', 'weight_type', 'RA2', 'DEC2']:
+        v = locals()[k]
+        if v is not None:
+            kwargs[k] = v
+
+    integer_isa = translate_isa_string_to_enum(isa)
+    rbinfile, delete_after_use = return_file_with_rbins(binfile)
+    bin_refs = np.arange(1, maxbinref + 1)
+    if link_in_ra:
+        bin_ref_perms = itertools.product(bin_refs, bin_refs)
+    else:
+        bin_ref_perms = [(1, binref) for binref in bin_refs]
```

Comment:
It is bothering me that we are using (and returning) a valid value for `nRA` (i.e., something that would work with `link_in_ra=True`) when *any* value should be allowed. However, passing negative values barfs up at the python wrapper level, while passing `nRA = 0` (via `bin_ref_perms = [(0, binref) for binref in bin_refs]`) causes an error message from the C implementation in [these lines](https://github.com/manodeep/Corrfunc/blob/master/mocks/DDtheta_mocks/countpairs_theta_mocks_impl.c.src#L548-L556). To me, `nRA=0` more accurately reflects that **no** RA gridding is being done.

The solution would be to use the `nRA=0` in this script here and update [these lines](https://github.com/manodeep/Corrfunc/blob/master/mocks/DDtheta_mocks/countpairs_theta_mocks_impl.c.src#L548-L556) to:

```C
    /* Only check the ra and dec bin refine factors (not all 3 bin refs)                                                                                               
       As evidenced by the PR #216, the error-message and resetting is not quite right! */
    if(options->link_in_ra && options->bin_refine_factors[0] < 1) {
        fprintf(stderr,"Warning: Linking in RA is requested, so the RA-bin refine factor *must* be >=1. Instead found bin refine factor =%d...resetting\n",
                options->bin_refine_factors[0]);
        reset_bin_refine_factors(options);
    }

    if(options->link_in_dec && options->bin_refine_factors[1] < 1) {
        fprintf(stderr,"Warning: Linking in DEC is requested, so the DEC-bin refine factor *must* be >=1. Instead found bin refine factor =%d...resetting\n",
                options->bin_refine_factors[1]);
        reset_bin_refine_factors(options);
    }
```

@kakirastern Are you happy to make these two changes? If not, I am completely happy to put those edits in myself.

Reply:
@manodeep  Yup, I would like to take on the task.

Reply:
@kakirastern - thanks!